policy/release/tasks_test.rego

```diff
@@ -0,0 +1,59 @@
+package policy.release.tasks
+
+import data.lib
+
+test_no_tasks_present {
+	lib.assert_equal(deny, {{
+		"code": "tasks_missing",
+		"msg": "No tasks found in PipelineRun attestation",
+		"effective_on": "2022-01-01T00:00:00Z",
+	}} | all_tasks_missing_errors) with input.attestations as [{"predicate": {
+		"buildType": lib.pipelinerun_att_build_type,
+		"buildConfig": {"tasks": []},
+	}}]
+}
+
+test_all_required_tasks_not_present {
+	lib.assert_equal(deny, all_tasks_missing_errors) with input.attestations as [{"predicate": {
+		"buildType": lib.pipelinerun_att_build_type,
+		"buildConfig": {"tasks": [{}]},
+	}}]
+}
+
+test_all_but_one_required_task_not_present {
+	present := {{
+		"code": "tasks_required",
+		"msg": "Required task 'sanity-inspect-image' not found in the PipelineRun attestation",
+		"effective_on": "2022-01-01T00:00:00Z",
+	}}
+
+	lib.assert_equal(deny, all_tasks_missing_errors - present) with input.attestations as [{"predicate": {
+		"buildType": lib.pipelinerun_att_build_type,
+		"buildConfig": {"tasks": [{"name": "sanity-inspect-image"}]},
+	}}]
+}
+
+test_several_tasks_not_present {
+	present := {
+		{
+			"code": "tasks_required",
+			"msg": "Required task 'sanity-inspect-image' not found in the PipelineRun attestation",
+			"effective_on": "2022-01-01T00:00:00Z",
+		},
+		{
+			"code": "tasks_required",
+			"msg": "Required task 'clamav-scan' not found in the PipelineRun attestation",
+			"effective_on": "2022-01-01T00:00:00Z",
+		},
+		{
+			"code": "tasks_required",
+			"msg": "Required task 'add-sbom-and-push' not found in the PipelineRun attestation",
+			"effective_on": "2022-01-01T00:00:00Z",
+		},
```

Comment:
Could maybe write a rule and make it tidier, e.g.:

    present := {
       mock_deny('sanity-inspect-image'),
       mock-deny('clamav-scan'),
       ...
    }

Or even something that takes a list, e.g.:

    present := mock_task_required_denies(['sanity-inspect-image','clamav-scan','add-sbom-and-push')